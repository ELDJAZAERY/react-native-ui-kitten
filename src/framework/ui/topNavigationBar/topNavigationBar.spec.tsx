import React from 'react';
import { View } from 'react-native';
import {
  render,
  fireEvent,
} from 'react-native-testing-library';
import {
  styled,
  StyleProvider,
  StyleProviderProps,
} from '@kitten/theme';
import {
  TopNavigationBar as TopNavigationBarComponent,
  Props as TopNavigationBarProps,
} from './topNavigationBar.component';
import * as config from './topNavigationBar.spec.config';

const TopNavigationBar = styled<TopNavigationBarComponent, TopNavigationBarProps>(TopNavigationBarComponent);

const Mock = (props?: TopNavigationBarProps): React.ReactElement<StyleProviderProps> => (
  <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
    <TopNavigationBar {...props} />
  </StyleProvider>
);

describe('@top-navigation-bar', () => {

  it('* title', () => {
    const component = render(<Mock title='Test'/>);
    expect(component).toMatchSnapshot();
  });

  it('* appearance: subtitle', () => {
    const component = render(<Mock appearance='subtitle' title='Test' subtitle='Subtitle'/>);
    expect(component).toMatchSnapshot();
  });

  it('* appearance: title-centered', () => {
    const component = render(<Mock appearance='title-centered' title='Test'/>);
    expect(component).toMatchSnapshot();
  });

  it('* appearance: title-centered-subtitle', () => {
    const component = render(<Mock appearance='title-centered-subtitle' title='Test' subtitle='Subtitle'/>);
    expect(component).toMatchSnapshot();
  });

  it('* actions checks', () => {
    const onLeftControl = jest.fn();
    const onRightControl1 = jest.fn();
    const onRightControl2 = jest.fn();
    const onRightControl3 = jest.fn();

    const component = render(
      <Mock
        appearance='title-centered-subtitle'
        title='Test'
        subtitle='Subtitle'
        leftControl={<View style={{ width: 20, height: 20, backgroundColor: 'white' }}/>}
        rightControls={[
          <View style={{ width: 20, height: 20, backgroundColor: 'white' }}/>,
          <View style={{ width: 20, height: 20, backgroundColor: 'red' }}/>,
          <View style={{ width: 20, height: 20, backgroundColor: 'green' }}/>,
        ]}
        onLeftControl={onLeftControl}
        onRightControls={[onRightControl1, onRightControl2, onRightControl3]}
      />,
    );

    expect(component).toMatchSnapshot();

    fireEvent.press(component.getByTestId('@top-navbar-left'));
    fireEvent.press(component.getByTestId('@top-navbar-right-0'));
    fireEvent.press(component.getByTestId('@top-navbar-right-1'));
    fireEvent.press(component.getByTestId('@top-navbar-right-2'));

    expect(onLeftControl).toHaveBeenCalled();
    expect(onRightControl1).toHaveBeenCalled();
    expect(onRightControl2).toHaveBeenCalled();
    expect(onRightControl3).toHaveBeenCalled();
  });

});
