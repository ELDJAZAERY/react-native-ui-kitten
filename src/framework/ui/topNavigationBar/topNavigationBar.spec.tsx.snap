// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@top-navigation-bar * actions checks 1`] = `
<View
  appearance="title-centered-subtitle"
  dispatch={[Function]}
  leftControl={
    <View
      style={
        Object {
          "backgroundColor": "white",
          "height": 20,
          "width": 20,
        }
      }
    />
  }
  onLeftControl={[MockFunction]}
  onRightControls={
    Array [
      [MockFunction],
      [MockFunction],
      [MockFunction],
    ]
  }
  rightControls={
    Array [
      <View
        style={
          Object {
            "backgroundColor": "white",
            "height": 20,
            "width": 20,
          }
        }
      />,
      <View
        style={
          Object {
            "backgroundColor": "red",
            "height": 20,
            "width": 20,
          }
        }
      />,
      <View
        style={
          Object {
            "backgroundColor": "green",
            "height": 20,
            "width": 20,
          }
        }
      />,
    ]
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "backgroundColor": "#3366FF",
        "height": 46,
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 4,
      },
      undefined,
    ]
  }
  subtitle="Subtitle"
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "#3366FF",
      "height": 46,
      "leftControlColor": "white",
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 4,
      "rightControlsColors": Array [
        "white",
        "white",
      ],
      "subtitleColor": "white",
      "subtitleFontSize": 12,
      "subtitleFontWeight": "400",
      "titleCentered": true,
      "titleColor": "white",
      "titleFontSize": 16,
      "titleFontWeight": "600",
      "withSubtitle": true,
    }
  }
  title="Test"
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="@top-navbar-left"
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "height": 20,
            "width": 20,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Test
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 12,
          "fontWeight": "400",
        }
      }
    >
      Subtitle
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="@top-navbar-right-0"
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "height": 20,
            "marginRight": 5,
            "width": 20,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="@top-navbar-right-1"
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "height": 20,
            "marginRight": 5,
            "width": 20,
          }
        }
      />
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="@top-navbar-right-2"
    >
      <View
        style={
          Object {
            "backgroundColor": undefined,
            "height": 20,
            "width": 20,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`@top-navigation-bar * appearance: subtitle 1`] = `
<View
  appearance="subtitle"
  dispatch={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "backgroundColor": "#3366FF",
        "height": 46,
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 4,
      },
      undefined,
    ]
  }
  subtitle="Subtitle"
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "#3366FF",
      "height": 46,
      "leftControlColor": "white",
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 4,
      "rightControlsColors": Array [
        "white",
        "white",
      ],
      "subtitleColor": "white",
      "subtitleFontSize": 12,
      "subtitleFontWeight": "400",
      "titleCentered": false,
      "titleColor": "white",
      "titleFontSize": 16,
      "titleFontWeight": "600",
      "withSubtitle": true,
    }
  }
  title="Test"
>
  <View
    style={
      Array [
        null,
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-start",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Test
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 12,
          "fontWeight": "400",
        }
      }
    >
      Subtitle
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
        },
        null,
      ]
    }
  />
</View>
`;

exports[`@top-navigation-bar * appearance: title-centered 1`] = `
<View
  appearance="title-centered"
  dispatch={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "backgroundColor": "#3366FF",
        "height": 46,
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 4,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "#3366FF",
      "height": 46,
      "leftControlColor": "white",
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 4,
      "rightControlsColors": Array [
        "white",
        "white",
      ],
      "subtitleColor": "white",
      "subtitleFontSize": 12,
      "subtitleFontWeight": "400",
      "titleCentered": true,
      "titleColor": "white",
      "titleFontSize": 16,
      "titleFontWeight": "600",
      "withSubtitle": false,
    }
  }
  title="Test"
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-start",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Test
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  />
</View>
`;

exports[`@top-navigation-bar * appearance: title-centered-subtitle 1`] = `
<View
  appearance="title-centered-subtitle"
  dispatch={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "backgroundColor": "#3366FF",
        "height": 46,
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 4,
      },
      undefined,
    ]
  }
  subtitle="Subtitle"
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "#3366FF",
      "height": 46,
      "leftControlColor": "white",
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 4,
      "rightControlsColors": Array [
        "white",
        "white",
      ],
      "subtitleColor": "white",
      "subtitleFontSize": 12,
      "subtitleFontWeight": "400",
      "titleCentered": true,
      "titleColor": "white",
      "titleFontSize": 16,
      "titleFontWeight": "600",
      "withSubtitle": true,
    }
  }
  title="Test"
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-start",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Test
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 12,
          "fontWeight": "400",
        }
      }
    >
      Subtitle
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  />
</View>
`;

exports[`@top-navigation-bar * title 1`] = `
<View
  appearance="default"
  dispatch={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      Object {
        "backgroundColor": "#3366FF",
        "height": 46,
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 4,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "backgroundColor": "#3366FF",
      "height": 46,
      "leftControlColor": "white",
      "paddingBottom": 12,
      "paddingHorizontal": 16,
      "paddingTop": 4,
      "rightControlsColors": Array [
        "white",
        "white",
      ],
      "subtitleColor": "white",
      "subtitleFontSize": 12,
      "subtitleFontWeight": "400",
      "titleCentered": false,
      "titleColor": "white",
      "titleFontSize": 16,
      "titleFontWeight": "600",
      "withSubtitle": false,
    }
  }
  title="Test"
>
  <View
    style={
      Array [
        null,
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-start",
        },
      ]
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Test
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
        },
        null,
      ]
    }
  />
</View>
`;
